src/NotificationEventMailing.php

```diff
@@ -271,20 +271,43 @@ public static function send(array $data)
                                 $doc->getFromDB($docID);
 
                                 //find width
-                                $width = null;
+                                $custom_width = null;
                                 if (preg_match("/width=[\"|'](\d+)(\.\d+)?[\"|']/", $matches[0][$pos], $wmatches)) {
-                                    $width = intval($wmatches[1]);
+                                    $custom_width = intval($wmatches[1]);
                                 }
-                                $height = null;
+                                $custom_height = null;
                                 if (preg_match("/height=[\"|'](\d+)(\.\d+)?[\"|']/", $matches[0][$pos], $hmatches)) {
-                                    $height = intval($hmatches[1]);
+                                    $custom_height = intval($hmatches[1]);
                                 }
 
+                                $img_infos  = getimagesize(GLPI_DOC_DIR . "/" . $doc->fields['filepath']);
+                                $initial_width = $img_infos[0];
+                                $initial_height = $img_infos[1];
+
+                                if ($custom_width && $custom_height) {
+                                    //compute height if need
+                                    if ($custom_width && is_null($custom_height)) {
+                                        $custom_height = $initial_height * $custom_width / $initial_width;
+                                    }
+
+                                    //compute width if needed
+                                    if ($custom_height && is_null($custom_width)) {
+                                        $custom_width = $initial_width * $custom_height / $initial_height;
+                                    }
+
+                                    //if both are null keep initial size
+                                    if (is_null($custom_height) && is_null($custom_width)) {
+                                        $custom_width = $initial_width;
+                                        $custom_height = $initial_height;
+                                    }
+                                }
+
```

Comment:
1. Prefer using `=== null` instead of `is_null()`. The is no need to use a function for this.
2. Using elseif will prevent to evaluate conditions when previous conditions matched.
3. Last block was never reached as all this piece of code was encapsualted in `if ($custom_width && $custom_height) { }`.
```suggestion
                                if ($custom_width !== null && $custom_height === null) {
                                    //compute height if needed
                                    $custom_height = $initial_height * $custom_width / $initial_width;
                                } elseif ($custom_height !== null && $custom_width === null) {
                                    //compute width if needed
                                    $custom_width = $initial_width * $custom_height / $initial_height;
                                } elseif ($custom_height === null && $custom_width === null) {
                                    //if both are null keep initial size
                                    $custom_width = $initial_width;
                                    $custom_height = $initial_height;
                                }
```